internal/config/tracing.go

```diff
@@ -0,0 +1,57 @@
+/**
+* Copyright 2018 Comcast Cable Communications Management, LLC
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package config
+
+import (
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+// TracingConfig provides the distributed tracing configuration
+type TracingConfig struct {
+	// Name is the name of the Tracing Config
+	Name string `toml:"-"`
+	// Implementation is the particular implementation to use. Ex OpenTelemetry.
+	// TODO generate with Rob Pike's Stringer
+	Implementation string `toml:"tracer_implementation"`
+	// Exporter is the format used to send to the collector
+	Exporter string `toml:"trace_exporter"`
+	// CollectorEndpoint is the URL of the trace collector it MUST be of Implementation implementation
+	CollectorEndpoint string `toml:"tracing_collector"`
+	// SampleRate sets the probability that a span will be recorded. Values between 0 and 1 are accepted.
+	SampleRate float64 `toml:"sample_rate"`
+
+	// Tracer is the actual Tracer Object associated with this configuration once loaded
+	Tracer trace.Tracer `toml:"-"`
+}
+
+// NewTracingConfig returns a new tracing config with default values
+func NewTracingConfig() *TracingConfig {
+	return &TracingConfig{
+		Name:           "default",
+		Implementation: defaultTracerImplemetation,
+		Exporter:       defaultExporterImplementation,
+		SampleRate:     1,
+	}
+}
+
+// Clone returns an exact copy of a tracing config
+func (tc *TracingConfig) Clone() *TracingConfig {
+	return &TracingConfig{
+		Name:              tc.Name,
+		Implementation:    tc.Implementation,
+		Exporter:          tc.Exporter,
+		SampleRate:        tc.SampleRate,
+		CollectorEndpoint: tc.CollectorEndpoint,
+		Tracer:            tc.Tracer,
```

Comment:
Since this is an interface we are just going to copy the pointer to the struct. Is that the expected behavior?

Reply:
Yeah, great point. I don't think that there is a guarantee that the impl _is_ a pointer.

Reply:
unfortunately, the Tracer interface does not provide a Clone() method that we can call, so this might have to do, short of re-registering the Tracer altogether during Clone. But since the configuration around the already-instantiated tracer would be identical to a new one that is registered, it should be fine to to re-use the existing Tracer during clone. If the user subsequently edits the cloned config, they would be on the hook for re-registering.